server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/rfc/ResponseObjectTest.scala

```diff
@@ -0,0 +1,199 @@
+        Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)}))
+      val expectedInvocationJsArray: JsArray = Json.arr({
+        Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)})
+
+      Json.toJson[Invocation](invocation) must be(JsSuccess(expectedInvocationJsArray))
+    }
+  }
+
+  "Deserialize RequestObject" must {
+    "succeed " in {
+      ResponseObject.deserialize(
+        """
+          |{
+          |  "methodResponses": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ]
+          |  ],
+          |  "sessionState": "75128aab4b1b"
+          |}
+          |""".stripMargin) must be(
+        ResponseObject.ResponseObject(
+          sessionState = ResponseObject.SessionState("75128aab4b1b"),
+          methodResponses = Seq(Invocation(Json.parse(
+            """[ "Core/echo", {
+              |      "arg1": "arg1data",
+              |      "arg2": "arg2data"
+              |    }, "c1" ]
+              |""".stripMargin).as[JsArray]))))
+    }
+
+    "succeed with many Capability, methodCalls" in {
+      ResponseObject.deserialize(
+        """
+          |{
+          |  "sessionState": "75128aab4b1b",
+          |  "methodCalls": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ],
+          |    [ "Core/echo2", {
+          |      "arg3": "arg3data",
+          |      "arg4": "arg4data"
+          |    }, "c2" ]
+          |  ]
+          |}
+          |""".stripMargin) must be(
+        ResponseObject.ResponseObject(
+          sessionState = ResponseObject.SessionState("75128aab4b1b"),
+          methodResponses = Seq(Invocation(Json.parse(
+            """[ "Core/echo", {
+              |      "arg1": "arg1data",
+              |      "arg2": "arg2data"
+              |    }, "c1" ]
+              |""".stripMargin).as[JsArray].append(Invocation(Json.parse(
+            """[ "Core/echo", {
+              |      "arg1": "arg1data",
+              |      "arg2": "arg2data"
+              |    }, "c1" ]
+              |""".stripMargin).as[JsArray]).value)))))
+    }
+  }
+
+  "Serialize RequestObject" must {
+    "succeed " in {
+      val requestObject: ResponseObject.ResponseObject = ResponseObject.ResponseObject(
+        sessionState = ResponseObject.SessionState("75128aab4b1b"),
+        methodResponses = Seq(Invocation(Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin).as[JsArray])))
+      Json.prettyPrint(Json.toJson(requestObject)) must be(
```

Comment:
I find it  suspicious. Can we rather do JSON aware assertions?

Reply:
It wrong and i fixed it.